### DISCUSSION TITLE
Como instalar e vincular o WakaTime no seu perfil

### DISCUSSION THREAD

**Discussion by MECR-0209** [community/community#115279] [Programming Help] [answered]
## Antes de começar a trabalhar no GitHub, siga estes passos:

- Acesse o site do WakaTime e faça seu cadastro em https://wakatime.com/.
- Após o cadastro, obtenha a chave de API do WakaTime.
- Integre o WakaTime com o seu ambiente de desenvolvimento, como o VS Code. Para isso, siga estas instruções:
  - No VS Code, vá para as extensões e instale o WakaTime.
  - Após a instalação siga as instruções da pagina, escreva algo em algum arquivo seu e reinicie o VS Code.
  - Vá ate o site do waka time e já vai aparecer que está vinculado, caso não apareça nada apenas espere um pouco.
- Na sua conta do WakaTime, você encontrará a chave de API que será utilizada posteriormente.

Após seguir esses passos, você estará pronto para utilizar o WakaTime integrado ao seu fluxo de desenvolvimento no GitHub.

### Só ir em cofigurações
![image](https://github.com/community/community/assets/137231429/c9e7e4a7-d298-4adc-a1cd-499621bb5cbf)

### Clique em conta e la esta a sua chave
![image](https://github.com/community/community/assets/137231429/bfc2fb34-4075-4b9d-9a2f-cde2ca2386b3)

# Passos para vincular as horas no seu perfil

# 1- Primeiro crie uma key do GitHub

### Va em configurações
![image](https://github.com/community/community/assets/137231429/c1e3ff0a-f3d0-40a4-b45d-e726c92813b2)

### Clique na ultima opção
![image](https://github.com/community/community/assets/137231429/64b3723b-1049-46dd-bb3d-40eacd660971)

![image](https://github.com/community/community/assets/137231429/37d7aaab-effb-44ac-9203-aa778386b90c)

![image](https://github.com/community/community/assets/137231429/e1ceba43-0516-4fd4-ac02-b64b3e270ba9)

### Vi em alguns lugares que colocam permissão em apenas algumas dessas opções, mas decidi permitir tudo no meu então não sei dizer bem qual permissão exata você precisa.

![image](https://github.com/community/community/assets/137231429/54d2798b-2150-4f02-8ce0-41537a4e20cd)

### No seu repositorio README va em configurações
![image](https://github.com/community/community/assets/137231429/f2fce60b-1347-4a5c-96af-b336831110fe)

### E então abre o seu secrets
![image](https://github.com/community/community/assets/137231429/c5759bf3-04fb-4784-b9c8-0c0f848e1d04)

### Lá você coloca o a sua chave o GitHub que você acabou de copiar e a chave do WakaTime
![image](https://github.com/community/community/assets/137231429/d7bb08be-b743-474f-bca3-002f0dd8a51d)

## 3- No seu arquivo README copie e cole isso no seu código: 

```
 <!--START_SECTION:waka-->
<!--END_SECTION:waka-->
```

### apenas crie um arquivo novo 

![image](https://github.com/community/community/assets/137231429/0b6e13c0-16f7-4a3e-8b8c-40a35da677da)

### e logo depois apenas digite o seguinte: /.github/workflows/nomeque_desejar.yml

![image](https://github.com/community/community/assets/137231429/903284ab-3fad-44d9-acae-d823823990ac)

### Você acabou de criar um workflow e um arquivo yml onde vai colocar esse código:

```
name: Work Stats Readme

on:
  workflow_dispatch:
  schedule:
    # Runs every 2 hours
    - cron: "0 */2 * * *"

jobs:
  update-readme:
    name: Update this repo's README
    runs-on: ubuntu-latest
    steps:
      - uses: athul/waka-readme@master 
        with:
          WAKATIME_API_KEY: ${{ secrets.WAKATIME_API_KEY }}
```

##  Vá nas configurações do seu repositório README, clique em ACTIONS e em GENERAL.
![image](https://github.com/community/community/assets/137231429/0ea5308e-f9c9-41a4-b7c1-ad7497fa8259)

### Um pouquinho abaixo na página tem essas opções, deixe como está na imagem e clique em SAVE
![image](https://github.com/community/community/assets/137231429/bd11d3c0-dd4a-4657-b73b-e9612876565f)

# Agora tente rodar o seu workflow.

## No seu repositório, vá em ACTIONS.
![image](https://github.com/community/community/assets/137231429/fcd82ae8-9e14-44b3-a47f-39ea881bca21)

## Logo depois, clique no seu workflow e clique em run.
![image](https://github.com/community/community/assets/137231429/714c8b8a-0004-442f-8eee-ac28639ce500)
![image](https://github.com/community/community/assets/137231429/18ef1e6c-3515-4db3-9c71-c2709ef00d01)

## A principio não é para dar um erro apos todos esses passos, caso aconteça algo deixa o comentario aqui.

**Comment by Akash1134** [comment, marked as answer]
Hi @MECR-0209 , thanks for participating!

We take our [Code of Conduct](https://github.com/community/community/blob/0ca37cf594e438fd4aec3c7f3736b95e6b0b3567/CODE_OF_CONDUCT.md) very seriously and want to help ensure that everyone has a good experience free of antagonism and harassment. Unfortunately, we don’t currently have moderators for languages other than English. Until that changes, we need to ask that everyone use English here in the GitHub Community when posting. We’ll be locking any posts in languages other than English for now, including this one.
